cloud/docker/docker_image.py

```diff
@@ -1,457 +1,502 @@
+        self.repository = parameters.get('repository')
+        self.rm = parameters.get('rm')
+        self.state = parameters.get('state')
+        self.tag = parameters.get('tag')
+        self.http_timeout = parameters.get('http_timeout')
+
+        if self.state in ['present', 'build']:
+            self.present()
+        elif self.state == 'absent':
+            self.absent()
+
+    def fail(self, msg):
+        self.client.fail(msg)
+
+    def present(self):
+        '''
+        Handles state = 'present', which includes building, loading or pulling an image,
+        depending on user provided parameters.
+
+        :returns None
+        '''
+        image = self.client.find_image(name=self.name, tag=self.tag)
+
+        if not image or self.force:
+            if self.path:
+                # build the image
+                params = dict(
+                    path=self.path,
+                    tag=self.name,
+                    rm=self.rm,
+                    nocache=self.nocache,
+                    stream=True,
+                    timeout=self.http_timeout,
+                    pull=self.pull,
+                    forcerm=self.rm,
+                    dockerfile=self.dockerfile,
+                    decode=True
+                )
+                if self.tag:
+                    params['tag'] = "%s:%s" % (self.name, self.tag)
+                if self.container_limits:
+                    params['container_limits'] = self.container_limits,
+                self.log("Building image %s" % (params['tag']))
+                if not self.check_mode:
+                    self.results['actions'].append("Built image %s from %s" % (params['tag'], self.path))
+                    for line in self.client.build(**params):
+                        self.log(line, pretty_print=True)
+                self.results['changed'] = True
+                image = self.client.find_image(name=self.name, tag=self.tag)
+                if image:
+                    self.results['image'] = image
+
+            elif self.load_path:
+                # Load the image from an archive
+                if not os.path.isfile(self.load_path):
+                    self.fail("Error loading image %s. Specified path %s does not exist." % (self.name,
+                                                                                             self.load_path))
+                name = self.name
+                if self.tag:
+                    name = "%s:%s" % (self.name, self.tag)
+
+                if not self.check_mode:
+                    self.results['actions'].append("Loaded image %s from %s" % (name, self.load_path))
+                    try:
+                        self.log("Reading image data from %s" % (self.load_path))
+                        image_tar = open(self.load_path, 'r')
+                        image_data = image_tar.read()
+                        image_tar.close()
+                    except Exception, exc:
```

Comment:
since this is already not 2.4 compatible, use py3 'friendly' exceptions:

`Exception as exc:`